consensus/src/common.rs

```diff
@@ -301,7 +301,10 @@ impl Consensus for ProtocolStateChain {
             // ring shift
             while shift_count > 0 {
                 rel_sub_window = (rel_sub_window + 1) % self.config().sub_windows_per_window.0;
-                projected_window[rel_sub_window as usize] = Length(0);
+                match projected_window.get_mut(rel_sub_window as usize) {
```

Comment:
Was this triggered because of the adversarial test cases?

Reply:
yes